pkg/cortex/cortex.go

```diff
@@ -109,7 +109,7 @@ func (c *Config) RegisterFlags(f *flag.FlagSet) {
 
 	c.Ruler.RegisterFlags(f)
 	c.ConfigDB.RegisterFlags(f)
-	c.ConfigStore.RegisterFlags(f)
+	c.ConfigStore.RegisterFlagsWithPrefix("alertmanager.", f)
```

Comment:
changing flag names is a breaking change for existing users. Is this really necessary to change? If it is, can we please call it out in the changelog so the release owner can highlight this fact in the appropriate release notes?

Reply:
The actual flags don't change in this case: https://github.com/cortexproject/cortex/pull/1571/files#diff-e8adc6214ca82bcbffe97a18e4753bc3L29

Only the functions in which they are registered, so this should be a noop for users.

Reply:
found the deprecated flags later in the review... resolving